### PR TITLE
Add Tvmult to Petsc PreconditionJacobi

### PR DESCRIPTION
**Pull request by dangars**
This adds the operation Tvmult to `PETScWrappers::PreconditionJacobi` as discussed in https://github.com/dealii/dealii/issues/8482#issuecomment-584555393

### CODE REVIEW

**Comment by dangars** [comment]
Should I add Tvmult to `PETScWrappers::PreconditionerBase` as a virtual function and then overload the function in `PreconditionJacobi` or I just add the function Tvmult to `PETScWrappers::PreconditionJacobi`?

**Comment by tjhei** [comment]
Can you implement https://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/PC/PCApplyTranspose.html#PCApplyTranspose in the base class instead? I would hope that that would work just as well.

File: tests/petsc/preconditioner_tvmult_01.cc [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+int
+main(int argc, char **argv)
+{
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+  MPILogInitAll                    log;
+
+  test<PETScWrappers::PreconditionJacobi>();
+  test<PETScWrappers::PreconditionBlockJacobi>();
+  test<PETScWrappers::PreconditionBoomerAMG>();
+  test<PETScWrappers::PreconditionNone>();
```
Comment (dangars) [inline tests/petsc/preconditioner_tvmult_01.cc:103]: I added several preconditioners to the tests. Not all the PETSc preconditioners support `PCApplyTranspose`. For example `Eisenstat` and `ParaSails` don't support it. If I try to run `Tvmult` with `Eisenstat` or `ParaSails` I obtain:
```
deal.II encountered an error while calling a PETSc function.
The description of the error provided by PETSc is "No support for this operation for this object type".
The numerical value of the original error code is 56.
```
I think that the message is quite clear. I think that there is no need for additional Asserts.

Comment (tjhei) [reply tests/petsc/preconditioner_tvmult_01.cc:103]: I agree.

**Comment by tjhei** [comment]
perfect, this looks good!

**Pull request merged**